pkg/provision/sync/sync.go

```diff
@@ -70,6 +74,7 @@ func createObjectGeneric(specObj crclient.Object, api ClusterAPI) error {
 	err := api.Client.Create(api.Ctx, specObj)
 	switch {
 	case err == nil:
+		api.Logger.Info("Created object", "kind", reflect.TypeOf(specObj).Elem().String(), "name", specObj.GetName())
 		return NewNotInSync(specObj, CreatedObjectReason)
```

Comment:
That seems not 100% clear. I don't return to the prev commit where it's introduced and leave thoughts here.
Initially, I understood NewNotInSync (as new object is created, was not in sync).
Then I iterate got the idea but then I checked NotInSyncError:
```go
// NotInSyncError is returned when a spec object is out-of-sync with its cluster counterpart
```

Actually, it may be more precise to specify that is was outdated but then we did a try to make it up to date.
That that's true for `UpdatedObjectReason`, `CreatedObjectReason`, `DeletedObjectReason` but not for `NeedRetryReason`...
Then maybe `NotInSyncError` should not be an error but just reporting the Sync state? and only `UnrecoverableSyncError` is an error that can become SyncError.

Or always  return `SyncState(oldClusterObject, updateClusterObject, madeOperation(create,delete,update,recreate), error)`


Then maybe it would be easier to have a dedicated func to create each of state? `NewObjectCreatedSyncState`, `NewObjectUpdatedSyncState`...

Let me know if it makes sense or not, or clarification is needed, I just put thoughts in not well-organized format )